### DISCUSSION TITLE
Error with applying reference file for genetic ancestry

### DISCUSSION THREAD

**Discussion by jasamack** [PGScatalog/pgsc_calc#234] [Q&A] [answered]
Hi, 

It's been very straightforward to apply pgsc_calc without accounting for genetic ancestry, but I have been encountering errors when I add --run_ancestry to my command.

- I downloaded the 1000G+HGDP reference file: https://ftp.ebi.ac.uk/pub/databases/spot/pgs/resources/pgsc_HGDP+1kGP_v1.tar.zst 
- I am using the latest version of the pipeline, v2.0.0-alpha.4.
- I am running the program within the Cambridge HPC system, relying on conda.

I am receiving an error that there are no variants that intersect between the reference and my target sample. I have attached my log. 

I appreciate any guidance on what could be going wrong.
[nextflow.log.txt](https://github.com/PGScatalog/pgsc_calc/files/14070139/nextflow.log.txt)

**Comment by smlmbrt** [comment, marked as answer]
Hi @jasamack - it seems like something is going wrong with the pvar/sorting? I also use the Cambridge HPC but have never used conda (only singularity). If you could try my singularity config that might sort it? You would use -c in nextflow and point to a file with similar contents (adapted for your needs):
```
prams {
  config_profile_description = 'Cambridge HPC cluster profile.'
  config_profile_contact = 'Sam Lambert'
  config_profile_url = "https://docs.hpc.cam.ac.uk/hpc"
}
singularity {
  enabled = true
  autoMounts = true
  cacheDir = "$HOME/rds/hpc-work/cache_nxf_singularity"
}
process {
    executor = 'slurm'
        clusterOptions = '-p icelake-himem -A [YOUR ACCOUNT, CHANGE]'
        withLabel:process_medium {
                cpus   = 4
                memory = 100.GB
                time   = 8.h
        }
        withName:INTERSECT_THINNED {
                scratch = false
        }
}
params {
        max_memory = 100.GB
        max_cpus = 12
        max_time = 12.h
}
```

It could be the `scratch = False `the fixes it.... I'm not sure why I did that (perhaps an out-of-memory problem I had once).

**Reply by jasamack** [reply]
Hi @smlmbrt,

Thanks so much for your help. I switched to singularity with your modification and was able to get past the sorting issue. After rerunning, I was receiving this error with FRAPOSA:

```
Jan-29 13:08:17.508 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'PGSCATALOG_PGSCALC:PGSCALC:ANCESTRY_PROJECT:FRAPOSA_PROJECT (PGSPOPsFetal)'

Caused by:
  Process `PGSCATALOG_PGSCALC:PGSCALC:ANCESTRY_PROJECT:FRAPOSA_PROJECT (PGSPOPsFetal)` terminated with an error exit status (1)

Command executed:

  fraposa GRCh38_reference_ALL_extracted         --method oadp         --dim_ref 10         --stu_filepref GRCh38_PGSPOPsFetal_ALL_oriented         --stu_filt_iid <(cut -f1 splitfamaa)         --out GRCh38_PGSPOPsFetal_ALL_oriented_splitfamaa
  
  cat <<-END_VERSIONS > versions.yml
  FRAPOSA_PROJECT:
      fraposa: TODO
  END_VERSIONS

Command exit status:
  1

Command output:
  FRAPOSA started.
  Reference data: GRCh38_reference_ALL_extracted
  Study data: GRCh38_PGSPOPsFetal_ALL_oriented
  Output prefix: GRCh38_PGSPOPsFetal_ALL_oriented_splitfamaa
  Method: oadp
  Reference dimension: 10
  Study dimension: 20
  Online SVD dimension: 40
  2024-01-29 13:08:12.774600
  Attemping to load saved reference PCA result...
  Warning: If you have changed the parameter settings, please delete GRCh38_reference_ALL_extracted_*.dat and GRCh38_reference_ALL_extracted.pcs then rerun FRAPOSA.
  Reference PCA result successfully loaded.
  2024-01-29 13:08:13.534661
  Loading study data...
  ERROR: 0 / 3486 ids in filter list match the study dataset
```

To fix the error, I edited the fraposa_project.nf module to modify the ID file to reflect both the FID and IID columns for my sample: 
`cut -f2 splitfamaa`

After this fix, I was receiving an error related to this issue: https://github.com/PGScatalog/pgsc_calc/issues/235. I edited the configuration file to include my appropriate paths, and I have the ancestry-adjusted score files now.

Thank you!

**Reply by smlmbrt** [reply]
Thank you for reporting the problem with FID, I opened an issue (#238) to try and fix that so it works out-of-the-box in future versions.

**Reply by cearring** [reply]
Hi, I am having the same error with FRAPOSA with the ids in the filter list not matching the study dataset.  I don't understand the fix that @jasamack shared unfortunately. I am working on a HPC using singularity profile and the same reference dataset.  When I opened up the reference data and study data file (extracted and oriented) I confirmed they do not have the same sample IDs but I don't want to go in and manually change them.  I am also a little confused why they are supposed to match up since they are from distinct studies...  
I appreciate any guidance on what could be going wrong.
[nextflow.log](https://github.com/user-attachments/files/16090032/nextflow.log)

**Reply by smlmbrt** [reply]
@cearring I think we're about to push a fix for issues where the IDs might misalign, stay tuned for the next release.

**Reply by smlmbrt** [reply]
Plink files with FID information (and/or duplicated IIDs) are now handled automatically after [`v2.0.0-beta.2`](https://github.com/PGScatalog/pgsc_calc/releases/tag/v2.0.0-beta.2).
